### PR TITLE
refactor workflow table state

### PR DESCRIPTION
**Pull request by panentheos**
This was a preliminary refactor in support of eventually showing launched job configurations, but it's valuable on its own. It does 2 things:
* Separates the workflow i/o table into its own component.
* Separates the state that drives it into its 3 parts: i/o definitions, config values, and errors.

This simplifies state handling a bit, and will also make it easy to render the same table in another context, e.g. the job config page, where we'll have i/o defs and config, but not errors.

### CODE REVIEW

File: src/pages/workspaces/workspace/tools/WorkflowView.js [hunk 1]
```diff
@@ -46,7 +56,76 @@ const styles = {
     position: 'absolute',
     bottom: '0.5rem',
     right: sideMargin
-  }
+  },
+  cell: optional => ({
+    fontWeight: !optional && 500,
```
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/tools/WorkflowView.js:61]: I've had issues with react complaining about `false` as a prop value, is that not a problem here?

**Review by MattPutnam — APPROVED** [review]

File: src/pages/workspaces/workspace/tools/WorkflowView.js [hunk 2]
```diff
@@ -233,60 +289,14 @@ class WorkflowViewContent extends Component {
...
-            ]
-          })
-        }
-      ])
+      h(WorkflowIOTable, {
+        which: key,
+        inputsOutputs,
+        config: modifiedConfig,
+        errors,
+        onChange: canCompute ? ((name, v) => this.setState(_.set(['modifiedConfig', key, name], v))) : undefined,
```
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/tools/WorkflowView.js:297]: Do we prefer the array for `set`, or ```modifiedConfig.${key}.${name}```?

**Review by zarsky-broad — APPROVED** [review]
Didn't run, but code looks solid.

File: src/pages/workspaces/workspace/tools/WorkflowView.js (hunk 1 above)
Comment (panentheos) [reply src/pages/workspaces/workspace/tools/WorkflowView.js:61]: I just checked, it seems to be ok here.

File: src/pages/workspaces/workspace/tools/WorkflowView.js (hunk 2 above)
Comment (panentheos) [reply src/pages/workspaces/workspace/tools/WorkflowView.js:297]: I usually use an array, to avoid dealing with string concatenation.

**Pull request merged**